### PR TITLE
feat(protocol): remove ERC20SnapshotUpgradeable from TaikoToken and BrigedERC20 tokens

### PR DESCRIPTION
**Pull request by dantaik**
ERC20SnapshotUpgradeable is not used at all and was inherited by Taiko Token and BridgedERC20 with an incorrect assumption. ERC20SnapshotUpgradeable has extra gas overhead so it's better to remove it from the inheritance.

### CODE REVIEW

File: packages/protocol/contracts/L1/TaikoToken.sol [hunk 1]
```diff
@@ -2,18 +2,24 @@
 pragma solidity 0.8.24;
 
 import "@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
-import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/ERC20SnapshotUpgradeable.sol";
```
Comment (dantaik) [inline packages/protocol/contracts/L1/TaikoToken.sol:5]: ERC20SnapshotUpgradeable reserves 50 slots

**Review by davidtaikocha — APPROVED** [review]

File: packages/protocol/contracts/L1/TaikoToken.sol [hunk 2]
```diff
@@ -2,18 +2,24 @@
...
 import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/ERC20VotesUpgradeable.sol";
 import "../common/EssentialContract.sol";
 import "../common/LibStrings.sol";
 
+/// @notice TaikoToken was `EssentialContract, ERC20SnapshotUpgradeable, ERC20VotesUpgradeable`.
+/// We use this contract to take 50 more slots to remove `ERC20SnapshotUpgradeable` from the parent
+/// contract list.
+abstract contract EssentialContract_ is EssentialContract {
+    uint256[50] private __slots_previously_used_by_ERC20SnapshotUpgradeable;
+}
```
Comment (adaki2004) [inline packages/protocol/contracts/L1/TaikoToken.sol:14]: Maybe a 'todo' here to dont forget to remove before mainnet, and only here bc. of testnet compatibility  ?

File: packages/protocol/contracts/tokenvault/BridgedERC20.sol [hunk 3]
```diff
@@ -1,22 +1,22 @@
...
 
-import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/IERC20MetadataUpgradeable.sol";
-import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/ERC20SnapshotUpgradeable.sol";
 import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/ERC20VotesUpgradeable.sol";
 import "./LibBridgedToken.sol";
 import "./BridgedERC20Base.sol";
 
+/// @notice BridgedERC20 was `BridgedERC20Base, ERC20SnapshotUpgradeable, ERC20VotesUpgradeable`.
+/// We use this contract to take 50 more slots to remove `ERC20SnapshotUpgradeable` from the parent
+/// contract list.
```
Comment (adaki2004) [inline packages/protocol/contracts/tokenvault/BridgedERC20.sol:10]: also a todo here to remove maybe, as a friendly reminder ?

File: packages/protocol/contracts/L1/TaikoToken.sol (hunk 2 above)
Comment (dantaik) [reply packages/protocol/contracts/L1/TaikoToken.sol:15]: added a comment, but not a TODO.

**Review by adaki2004 — APPROVED** [review]

**Review by Brechtpd — APPROVED** [review]

**Pull request merged**